vault/expiration.go

```diff
@@ -668,22 +668,22 @@ func (m *ExpirationManager) revokePrefixCommon(prefix string, force, sync bool)
 	// Ensure there is a trailing slash; or, if there is no slash, see if there
 	// is a matching specific ID
 	if !strings.HasSuffix(prefix, "/") {
-		le, err := m.loadEntry(prefix)
+		le, err := m.loadEntry(ctx, prefix)
 		if err == nil && le != nil {
 			if sync {
-				if err := m.revokeCommon(m.quitContext, prefix, force, false); err != nil {
+				if err := m.revokeCommon(ctx, prefix, force, false); err != nil {
```

Comment:
This function may take a long time, but the user request will time out after 90 seconds (by default). I feel like when in sync mode should instead keep quitContext so that the function can keep running even after the client context is canceled.

Reply:
We put the quit context on the public methods (`RevokeForce`, `RevokePrefix`).  The incoming context to these methods is going to be the active context.